DOCS-13158-db-system-properties-oracle-kt (#1783)
mulesoft/docs-connectors: db/1.10/modules/ROOT/pages/database-connector-troubleshooting.adoc

In Oracle, an empty string equals to `NULL`, send `''` instead.

== Troubleshoot Oracle's XML Extensions using System Properties

> To use Oracle XML extensions, you need to add Oracle jar files as dependencies. One of these jar files is `xmlparserv2.jar` that exports XML parser capabilities via the Java SPI (Service Provider Interface). When internal parts of Mule use this parser capabilities, they can break some Mule components' expectations and cause XML parsing errors, such as the following:

Reviewer: ```suggestion
To use Oracle XML extensions, you must add Oracle jar files as dependencies. One of these jar files is `xmlparserv2.jar`, which exports XML parser capabilities via the Java SPI (Service Provider Interface). When internal parts of Mule use this parser, XML parsing errors such as the following:
```

[source,xml,linenums]
----
	<Line 43, Column 57>: XML-24509: (Error) Duplicated definition for: 'identifiedType' <Line 60, Column 28>: XML-24509: (Error) Duplicated definition for: 'beans' <Line 140, Column 34>: XML-24509: (Error) Duplicated definition for: 'description'
----

> One way of dealing with this issue is to add the `xmlparserv2.jar` file as an additional plugin dependency in the Database Connector, so the dependency is visible only from within the connector. Though, that won't work if you need to share an Oracle connection among modules, which is the case if you want to provide a DataSource as a Spring configured bean. A more general solution is to disable the XML parser SPI lookup by setting the following system properties, so Mule internal components always get the XML parser they expect:

Reviewer: Trying to tighten up the wording...

To deal with this issue, do either of the following:

* Add the  `xmlparserv2.jar` file as an additional plugin dependency in the Database Connector, so that the dependency is visible only from within the connector.

(indent this under the first bullet) If you need to share an Oracle connection among modules, which is the case if you want to provide a DataSource as a Spring configured bean, use the next solution. 

* Disable the XML parser SPI lookup by setting the following system properties, so Mule internal components always get the XML parser they expect:

.System Properties:
[source,xml,linenums]
----
-Djavax.xml.parsers.SAXParserFactory=com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl
-Djavax.xml.parsers.DocumentBuilderFactory=com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl
----

> These properties avoid any trouble if, for instance, you need to set up Oracle .jars as shared libraries.

Reviewer: Is this statement needed or does it repeat the statement earlier about sharing an Oracle connection among modules?

Author: Yes, it is needed because mentions the shared libraries.

> For further information about how to set system properties, refer to the xref:mule-runtime::mule-app-properties-system.adoc[system properties] documentation.

Reviewer: ```suggestion
For more information about how to set system properties, refer to the xref:mule-runtime::mule-app-properties-system.adoc[system properties] documentation.
```
